0000-account-abstraction.md

```diff
@@ -0,0 +1,246 @@
+
+- There should be a fee and appropriate spam protection to create or update account contructs.
+
+- A portion of the fee could be stored on creation and released if an account is destroyed to encourage not having accounts live forever.
+
+- Destroying an account is only possible for an account with no open positions and no liquidity commitments.
+
+- When destroying an account, an account ID to inherit any and all assets must be specified.
+  This may be an insurance pool, network treasury, etc. account.
+
+- Note: all the virtual account types have a "controller" concept that if specified can configure the account, including changing the controller (unless prevented by permissions, see below).
+  This could be genericised/abstracted and the same code/functionality reused in all cases.
+
+
+## Multisig account requirements
+
+Multisig accounts allow an action to be taken by more than one party/account and for actions to require more than one party's approval to happen.
+
+- A mutlisig can be created and is given an ID.
+  The account ID for a given multisig never changes once it is assigned.
+
+- A multisig contains a list of one of more accounts plus associated weights (default weight of 1 for all accounts if not specified).
+
+- A threshold integer number of approvals.
+  At least this number of the listed accounts must approve a transaction.
+  May be blank/zero in which as the number of accounts check automatically passes.
+
+- A threshold weight-fraction for approval.
+  At least this frsction of the sum of all  accounts' weights must approve a transaction .
+  May be blank/zero in which as the weight check automatically passes.
+
+- If both thresholds are zero than any of the listed accounts can always unilaterally take action.
+
+- A maximum transacton timeout in seconds optionally set at the multisig account level.
+  If present and non-zero then a transaction must be approved within this time or it is cancelled.
+
+- A maximum transaction timeout in seconds optionally set on each transaction.
+  If present and non-zero then a transaction must be approved within this time or it is cancelled.
+  Does not overried the account level timeout, i.e. the timeout applied if both are present is the shorter of the two.
+
+- A controller account ID.
+  The controller can be the multisig itself (self controlling) or any other account of any type.
+	If not present then the multisig is immutable once created.
+
+- The controller can change the controller account ID.
+
+- The controller can add to and remove from the list of accounts in the multisig.
+
+- The controller can update the weights of each account.
+
+- The controller can change the approval number threshold.
+
+- The controller can change the approcal weight-fraction threshold.
+
+- The controller can change the account-wide timeout.
+
+
+## Account permissions and restrictions
```

Comment:
See my comment on the PR @witgaw, I think it'd be good to descope this from this PR and include basic restrictions of scope of use of funds in the fund account spec directly.